index.bs

```diff
@@ -55,13 +55,326 @@ before a user has picked a location to save to, without forcing the website to u
+may correspond directly to a path on the local file system, or may not
+correspond to any file on disk at all. The actual physical location of the
+corresponding [=/file system entry=] is [=implementation-defined=].
+
+<p class=example id=example-path>Let |path| be the [=/list=]
+« "`data`", "`drafts`", "`example.txt`" ».
+There is no expectation that a file named `example.txt` exists anywhere on disk.
+
+A [=/file system path=] |a| is <dfn for="file system path">the same path as</dfn>
+a [=/file system path=] |b| if
+|a|'s [=list/size=] is the same as |b|'s [=list/size=] and
+[=list/for each=] |index| of |a|'s [=list/indices=]
+|a|.\[[|index|]] is |b|.\[[|index|]].
+
+<div algorithm>
+Each [=/file system=] has an associated
+<dfn for="file system">locate an entry</dfn>
+algorithm, which takes [=/file system path=] |path| and runs an
+[=implementation-defined=] series of steps returning a [=file system entry=] or null.
+A non-null return value |entry| must adhere to these constraints:
+
+- [=file system/Getting the path=] with |entry| returns |path|,
+  provided no intermediate file system operations were run.
+- |entry|'s [=file system entry/name=] is the last [=list/item=] of |path|.
+
+</div>
+
+<div algorithm>
+Each [=/file system=] has an associated
+<dfn for="file system">get the path</dfn>
+algorithm, which takes [=/file system entry=] |entry| and runs an
+[=implementation-defined=] series of steps returning a [=file system path=]. The
+return value |path| must adhere to these constraints:
+
+- [=file system/Locating an entry=] with |path| returns |entry|,
+  provided no intermediate file system operations were run.
+- |entry|'s [=file system entry/name=] is the last [=list/item=] of |path|.
+
+</div>
+Each [=/file system=] generates <dfn for="file system">file system event</dfn>s
+for actions that modifies its state.
+
+A [=file system/file system event=] has
+a {{FileSystemChangeType}} <dfn for="file system/file system event">type</dfn>,
+an <dfn for="file system/file system event">entry type</dfn> (a {{FileSystemHandleKind}} or null),
+a [=/file system path=] <dfn for="file system/file system event">modified path</dfn>,
+and a <dfn for="file system/file system event">from path</dfn> (a [=/file system path=] or null).
+The [=file system/file system event/from path=] must be set for when
+[=file system/file system event/type=] is "{{FileSystemChangeType/moved}}" and
+must be null otherwise.
+
+Note: The [=file system/file system event/entry type=] is null when the
+[=/file system=] cannot determine the {{FileSystemHandleKind}} of the
+[=file system/file system event/modified path=] due to the underlying
+[=file system entry=] no longer being there.
+
+A [=/file system=] has an associated [=/set=] of
+[=file system observations=] <dfn for="file system">observations</dfn>
+which is initialized to the empty set.
+
+When a [=/file system=] |fileSystem| fires a [=/list=] of
```

Comment:
Yeah I tried to set this change up so that we can eventually spec out how the BFS sends file system change events. Let me know if you want me to explore that in this PR.